### PR TITLE
Use flock to serialize calls (boo#1188500)

### PR DESCRIPTION
**Pull request by lnussel**

### CODE REVIEW

**Review by Vogtinator — APPROVED** [review]

**Review by Vogtinator — CHANGES_REQUESTED** [review]
Actually, the new dep on `flock` is a bit too much...
In the base container it needs:

```
The following 23 NEW packages are going to be installed:
  chkstat libaudit1 libblkid1 libcap-ng0 libcrypt1 libeconf0 libfdisk1 libmount1 libsemanage-conf libsemanage2 libsepol2 libsmartcols1 libutempter0 libuuid1 login_defs pam pam_unix permissions permissions-config shadow system-group-hardware
  sysuser-shadow util-linux

23 new packages to install.
Overall download size: 4.5 MiB. Already cached: 0 B. After the operation, additional 12.3 MiB will be used.
```

File: update-ca-certificates [hunk 1]
```diff
@@ -76,6 +76,12 @@ esac
 rm -f /etc/pki/trust/.updated
 
 mkdir -p "$statedir"
+if [ -z "$_CA_CERTIFICATES_LOCKED" ]; then
+    export _CA_CERTIFICATES_LOCKED="1"
```
Comment (Vogtinator) [inline update-ca-certificates:80]: Alternative to `flock`, but cleanup can fail:

```
    (set -o noclobber; while ! > /run/ucc.lock; do sleep 0.5; done) 2>/dev/null
    trap 'rm /run/ucc.lock' EXIT
```

Comment (lnussel) [reply update-ca-certificates:80]: too ugly for my taste

Comment (lnussel) [reply update-ca-certificates:80]: busybox has flock too

Comment (lnussel) [reply update-ca-certificates:80]: another alternative would be to call flock only if available. old behavior if not present which won't happen on a full system with util-linux

Comment (Vogtinator) [reply update-ca-certificates:80]: Would work too

Comment (lnussel) [reply update-ca-certificates:81]: updated

File: update-ca-certificates [hunk 2]
```diff
@@ -76,6 +76,13 @@ esac
 rm -f /etc/pki/trust/.updated
 
 mkdir -p "$statedir"
+# serialize calls if we can
+if [ -z "$_CA_CERTIFICATES_LOCKED" -a -x /usr/bin/flock ]; then
```
Comment (Vogtinator) [inline update-ca-certificates:80]: `-a` is deprecated

```suggestion
if [ -z "$_CA_CERTIFICATES_LOCKED" ] && [ -x /usr/bin/flock ]; then
```

Comment (lnussel) [reply update-ca-certificates:80]: fixing shellcheck complaints would be other commit :)

**Pull request merged**